# HG changeset patch
# User halton@judo
# Date 1197517323 -28800
# Node ID f6e296bee56ccd8245538124ccd575dd7d8341e0
# Parent  c03b2c8f36dd14e1befd26dcd8794430112e3076
Fix ld and ast problem on Solaris

diff -r c03b2c8f36dd -r f6e296bee56c ChangeLog
--- a/ChangeLog	Wed Dec 12 09:10:31 2007 -0600
+++ b/ChangeLog	Thu Dec 13 11:42:03 2007 +0800
@@ -1,3 +1,9 @@ 2007-09-13  "Daniel P. Berrange  <berran
+2007-10-30  Halton Huo <halton.huo@sun.com>
+
+	* src/Makefile.am: Use -export-symbols-regex to mark with symbols
+	to export instead of a GNU LD --version-script. This allows us to 
+	build on Solaris' linker.
+
 2007-09-13  "Daniel P. Berrange  <berrange@redhat.com>
 
 	* src/gvnc.c:
diff -r c03b2c8f36dd -r f6e296bee56c configure.ac
--- a/configure.ac	Wed Dec 12 09:10:31 2007 -0600
+++ b/configure.ac	Thu Dec 13 11:42:03 2007 +0800
@@ -85,6 +85,24 @@ AC_SUBST(GNUTLS_CFLAGS)
 AC_SUBST(GNUTLS_CFLAGS)
 AC_SUBST(GNUTLS_LIBS)
 
+case $host in
+  *-*-solaris* | *-*-sunos*)
+	AC_CHECK_LIB(ast, _ast_printf,
+      [AST_LIBS="-last"],
+      [
+        echo "Error! You need to have libast around."
+        exit -1
+      ])
+    AC_CHECK_HEADER(ast/endian.h,,
+      [AST_CFLAGS="-I/usr/include/ast"],
+      [
+        echo "Error! You need to have ast/endian.h around."
+        exit -1
+      ])
+	AC_SUBST(AST_CFLAGS)
+	AC_SUBST(AST_LIBS) ;;
+esac
+  
 if test "$WITH_PYTHON" = "yes"; then
   PKG_CHECK_MODULES(PYGTK, pygtk-2.0 >= $PYGTK_REQUIRED)
   AC_SUBST(PYGTK_CFLAGS)
diff -r c03b2c8f36dd -r f6e296bee56c src/Makefile.am
--- a/src/Makefile.am	Wed Dec 12 09:10:31 2007 -0600
+++ b/src/Makefile.am	Thu Dec 13 11:42:03 2007 +0800
@@ -1,13 +1,14 @@
 
-EXTRA_DIST = libgtk-vnc_sym.version vncmarshal.txt
+EXTRA_DIST = vncmarshal.txt
 
 lib_LTLIBRARIES = libgtk-vnc-1.0.la
 
-libgtk_vnc_1_0_la_LIBADD = @GTK_LIBS@ @GNUTLS_LIBS@
+libgtk_vnc_1_0_la_LIBADD = @GTK_LIBS@ @GNUTLS_LIBS@ @AST_LIBS@
 libgtk_vnc_1_0_la_CFLAGS = @GTK_CFLAGS@ @GNUTLS_CFLAGS@ @WARNING_CFLAGS@ \
                            -DSYSCONFDIR=\""$(sysconfdir)"\" \
-                           @DEBUG_CFLAGS@
-libgtk_vnc_1_0_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libgtk-vnc_sym.version \
+                           @DEBUG_CFLAGS@ \
+                           @AST_CFLAGS@
+libgtk_vnc_1_0_la_LDFLAGS = -export-symbols-regex 'vnc_display_' 'gvnc_' 'coroutine_' \
                             -version-info 0:1:0
 
 gtk_vnc_includedir = $(includedir)/gtk-vnc-1.0/
diff -r c03b2c8f36dd -r f6e296bee56c src/libgtk-vnc_sym.version
--- a/src/libgtk-vnc_sym.version	Wed Dec 12 09:10:31 2007 -0600
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,66 +0,0 @@
-{
-  global:
-    vnc_display_get_type;
-    vnc_display_credential_get_type;
-
-    vnc_display_new;
-    vnc_display_open_fd;
-    vnc_display_open_host;
-    vnc_display_is_open;
-    vnc_display_close;
-
-    vnc_display_send_keys;
-
-    vnc_display_set_credential;
-
-    vnc_display_set_use_shm;
-    vnc_display_set_pointer_local;
-    vnc_display_set_pointer_grab;
-    vnc_display_set_keyboard_grab;
-
-    vnc_display_get_pixbuf;
-
-    vnc_display_get_width;
-    vnc_display_get_height;
-    vnc_display_get_name;
-
-    vnc_display_client_cut_text;
-
-    gvnc_new;
-    gvnc_free;
-    gvnc_close;
-    gvnc_shutdown;
-
-    gvnc_open_fd;
-    gvnc_open_host;
-    gvnc_is_open;
-
-    gvnc_initialize;
-    gvnc_is_initialized;
-
-    gvnc_server_message;
-    gvnc_client_cut_text;
-    gvnc_pointer_event;
-    gvnc_key_event;
-    gvnc_framebuffer_update_request;
-    gvnc_set_encodings;
-    gvnc_set_pixel_format;
-    gvnc_set_shared_buffer;
-    gvnc_has_error;
-    gvnc_set_local;
-    gvnc_set_vnc_ops;
-    gvnc_shared_memory_enabled;
-    gvnc_get_name;
-    gvnc_get_width;
-    gvnc_get_height;
-
-    coroutine_init;
-    coroutine_release;
-    coroutine_swap;
-    coroutine_self;
-    coroutine_yieldto;
-    coroutine_yield;
-
-  local:
-      *;
-};
